pkgpanda/__init__.py

```diff
@@ -442,15 +452,48 @@ def validate_username(username):
         if not re.match(username_regex, username):
             raise ValidationError("Username must begin with `dcos_` and only have a-z and underscore after that")
 
-    def add_user(self, username):
+    @staticmethod
+    def validate_group(group):
+        # Empty group is allowed.
+        if not group:
+            return
+
+        UserManagement.validate_group_name(group)
+
+        try:
+            grp.getgrnam(group)
+        except KeyError:
+            raise ValidationError("Group {} does not exist on the system".format(group))
+
+    @staticmethod
+    def validate_group_name(group_name):
+        if not group_name:
+            return
+
+        if not re.match(linux_group_regex, group_name):
+            raise ValidationError("Group {} has invalid name, must match the following regex: {}".format(
+                group_name, linux_group_regex))
+
+    @staticmethod
+    def validate_user_group(username, groupname):
+        user = pwd.getpwnam(username)
+        if not groupname:
+            return
+
+        if user.pw_gid != grp.getgrnam(groupname).gr_gid:
```

Comment:
Almost ship, but users can have more than one group. They just have a "primary" group (Which is what this is). Should check that the group is in the set of groups to which the user belongs (Or that the group contains this user).